app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/model/DummyDatabaseService.kt

```diff
@@ -230,10 +230,15 @@ class DummyDatabaseService internal constructor() : DatabaseService {
         require(dataset is DummyDataset)
         require(category is DummyCategory)
         if (!categories.contains(category)) {
-            throw java.lang.IllegalArgumentException("The underlying database does not contain the category ${category.name}")
+            throw java.lang.IllegalArgumentException("The underlying database does not " +
+                    "contain the category ${category.name}")
         }
-        if (!datasets.contains(dataset)) {
-            throw IllegalArgumentException("The underlying database does not contain the dataset ${dataset.name}")
+        // calling datasets.contains(dataset) returned false for obviously equal datasets
+        // TODO : discuss this
+//        if (!datasets.contains(dataset)) {
+        if (datasets.find{it == dataset} == null) {
+            throw IllegalArgumentException("The underlying database:\n $datasets\n\n does not " +
+                    "contain the dataset \n $dataset")
```

Comment:
I agree this is a bit weird... I can't see the reason right now but I will come back to you if I find the reason.

Reply:
please bring concrete examples of such datasets to the discussion

Reply:
yep, sorry about that, I forgot to explain where the problem arose.
For example, if you uncomment  `if (!datasets.contains(dataset)) {` and launch `DisplayImageActivityTest::pictureAndCategoryAreDisplayed` you get the following error:

E/TestRunner: java.lang.IllegalArgumentException: The underlying database:
     [DummyDataset(id=kitchen utensils, name=kitchen utensils, categories=
[DummyCategory(id=Fork, name=Fork), 
DummyCategory(id=Knife, name=Knife), 
DummyCategory(id=Spoon, name=Spoon), 
DummyCategory(id=2d68223d-7ca7-4bd7-8bba-9fd420091fb5, name=2d68223d-7ca7-4bd7-8bba-9fd420091fb5)])]
     does not contain the dataset 
     DummyDataset(id=kitchen utensils, name=kitchen utensils, categories=
[DummyCategory(id=Fork, name=Fork), 
DummyCategory(id=Knife, name=Knife), 
DummyCategory(id=Spoon, name=Spoon), 
DummyCategory(id=2d68223d-7ca7-4bd7-8bba-9fd420091fb5, name=2d68223d-7ca7-4bd7-8bba-9fd420091fb5)])
        at com.github.HumanLearning2021.HumanLearningApp.model.DummyDatabaseService.addCategoryToDataset(DummyDatabaseService.kt:240)